asdf/tags/wcs/wcs.py

```diff
@@ -65,107 +65,10 @@ class StepType(dict, AsdfType):
 
 class FrameType(AsdfType):
     name = "wcs/frame"
-    requires = ['gwcs', 'astropy-1.3.3']
+    requires = ['gwcs']
     types = ['gwcs.Frame2D']
     version = '1.2.0'
 
-    import astropy
-    _astropy_version = astropy.__version__
-    # This indicates that Cartesian Differential is not available
-    _old_astropy = astropy.__version__ <= '1.3.3'
-
-    @classmethod
-    def _get_reference_frame_mapping(cls):
-        if hasattr(cls, '_reference_frame_mapping'):
-            return cls._reference_frame_mapping
-
-        from astropy.coordinates import builtin_frames
-
-        cls._reference_frame_mapping = {
-            'ICRS': builtin_frames.ICRS,
-            'FK5': builtin_frames.FK5,
-            'FK4': builtin_frames.FK4,
-            'FK4_noeterms': builtin_frames.FK4NoETerms,
-            'galactic': builtin_frames.Galactic,
-            'galactocentric': builtin_frames.Galactocentric,
-            'GCRS': builtin_frames.GCRS,
-            'CIRS': builtin_frames.CIRS,
-            'ITRS': builtin_frames.ITRS,
-            'precessed_geocentric': builtin_frames.PrecessedGeocentric
-        }
-
-        return cls._reference_frame_mapping
-
-    @classmethod
-    def _get_inverse_reference_frame_mapping(cls):
-        if hasattr(cls, '_inverse_reference_frame_mapping'):
-            return cls._inverse_reference_frame_mapping
-
-        reference_frame_mapping = cls._get_reference_frame_mapping()
-
-        cls._inverse_reference_frame_mapping = {}
-        for key, val in reference_frame_mapping.items():
-            cls._inverse_reference_frame_mapping[val] = key
-
-        return cls._inverse_reference_frame_mapping
-
-    @classmethod
-    def _reference_frame_from_tree(cls, node, ctx):
```

Comment:
So, as far as my limited understanding of all this can tell, all of this *should* be able to be deleted once astropy/astropy#7079 is in? (With the priviso of needing to support all the representation types in that PR?) is that base assumption correct @nden ?

Reply:
There's a question about whether we want to retain this here for backwards compatibility. In any case, these things are moving to the `gwcs` package, so it would be a question as to whether `gwcs` needs to continue to maintain them, or whether they can be dropped there.

(@nden please chime in if I'm wrong about this)

Reply:
As @drdavella said this needs to stay for backwards compatibility (though asdf versioning needs to be defined better).

gwcs uses the low level builtin frames in coordinates and I don't see them implemented in the astropy PR but may be this is coming @Cadair? In the future I think gwcs will use astropy's schemas for frames and coordinates.